examples_tests/31.SkinningDataBenchmark/main.cpp

```diff
@@ -4,396 +4,385 @@
-        //memcpy(_dst + i*size_mat4x3_dst, _src + i*4*7, size_mat4x3*sizeof(float));
-        for (size_t j = 0u; j < 4u; ++j)
-            memcpy(_dst + i*size_mat4x3_dst + 4*j, _src + i*4*7 + 3*j, 3*sizeof(float));
-        // each mat3 column aligned to 4*sizeof(float)
-        for (size_t j = 0u; j < 3u; ++j)
-            memcpy(_dst + offset_mat3_dst + i*size_mat3_dst + j*4, _src + i*4*7 + size_mat4x3_src + j*3, 3*sizeof(float));
+        auto disk = am->getGeometryCreator()->createDiskMesh(0.5f, tesselation[i]);
+        auto newIdxBuffer = am->getMeshManipulator()->idxBufferFromTrianglesFanToTriangles(indices.data(), disk.indexCount, EIT_16BIT);
+        disk.indexBuffer = { 0ull, newIdxBuffer };
+        disk.indexCount = newIdxBuffer->getSize() / sizeof(uint16_t);
+        disk.indexType = EIT_16BIT;
+        disk.assemblyParams.primitiveType = EPT_TRIANGLE_LIST;
+        createMeshBufferFromGeometryCreatorReturnData(disk, disks[i], i);
     }
 
-    const size_t perInstance = (size_mat3_dst + size_mat4x3_dst) * _boneCnt;
-
-    for (size_t i = 1u; i < _instCnt; ++i)
-        memcpy(_dst + i*perInstance, _dst, sizeof(float)*perInstance);
+    os::Printer::print("Disks creation done.\n");
 
-    return sizeof(float) * perInstance * _instCnt;
-}
-size_t convertBuf4(const float* _src, float* _dst, const size_t _boneCnt, const size_t _instCnt)
-{
-    const size_t size_mat4x3_src = 12u; // 12 floats
-    const size_t 
-        offset_mat4x3_0 = 0u,
-        offset_mat4x3_1 = _boneCnt*4,
-        offset_mat4x3_2 = _boneCnt*8,
-        offset_mat4x3_3 = _boneCnt*12,
-        offset_mat3_0 = _boneCnt*16,
-        offset_mat3_1 = _boneCnt*20,
-        offset_mat3_2 = _boneCnt*24;
-    for (size_t i = 0u; i < _boneCnt; ++i)
+        //pack disks
+    MeshPackerBase::PackedMeshBuffer<ICPUBuffer> packedMeshBuffer;
+    MeshPackerBase::PackedMeshBufferData mb;
     {
-        memcpy(_dst + offset_mat4x3_0 + i*4, _src + i*4*7 + 0*3, 3*sizeof(float));
-        memcpy(_dst + offset_mat4x3_1 + i*4, _src + i*4*7 + 1*3, 3*sizeof(float));
-        memcpy(_dst + offset_mat4x3_2 + i*4, _src + i*4*7 + 2*3, 3*sizeof(float));
-        memcpy(_dst + offset_mat4x3_3 + i*4, _src + i*4*7 + 3*3, 3*sizeof(float));
-        // each mat3 column aligned to 4*sizeof(float)
-        memcpy(_dst + offset_mat3_0 + i*4, _src + i*4*7 + size_mat4x3_src + 0*3, 3*sizeof(float));
-        memcpy(_dst + offset_mat3_1 + i*4, _src + i*4*7 + size_mat4x3_src + 1*3, 3*sizeof(float));
-        memcpy(_dst + offset_mat3_2 + i*4, _src + i*4*7 + size_mat4x3_src + 2*3, 3*sizeof(float));
-    }
+        auto allocParams = MeshPackerBase::AllocationParams();
+        allocParams.MDIDataBuffSupportedCnt = 1024;
+        allocParams.MDIDataBuffMinAllocSize = 512;
+        allocParams.indexBuffSupportedCnt = 1000 * 160 * 10;
+        allocParams.indexBufferMinAllocSize = 256;
+        allocParams.vertexBuffSupportedCnt = 1000 * 160 * 10;
+        allocParams.vertexBufferMinAllocSize = 256;
```

Comment:
you can probably optimize those values knowing the disk meshes you are packing. What about this shrinking issue you had yesterday?

Reply:
@crisspl no need, he should shrink the buffers later anyway, even if he overallocates the memory it should not impact the benchmarks ;)